chips/earlgrey/src/pinmux.rs

```diff
@@ -0,0 +1,208 @@
+            UP = 1,
+        ],
+        KEEPER_EN OFFSET(4) NUMBITS(1) [],
+        SCHMITT_EN OFFSET(5) NUMBITS(1) [],
+        OPEN_DRAIN_EN OFFSET(6) NUMBITS(1) [],
+        SLEW_RATE OFFSET(16) NUMBITS(2) [],
+        DRIVE_STRENGTH OFFSET(20) NUMBITS(4) [],
+    ],
+];
+
+type PadAttribute = LocalRegisterCopy<u32, PAD_ATTR::Register>;
+
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum Pad {
+    Mio(MuxedPads),
+    Dio(DirectPads),
+}
+
+impl Pad {
+    // Extract value of attributes using common layout
+    fn pad_attr(self) -> PadAttribute {
+        PadAttribute::new(match self {
+            Self::Mio(mio) => PINMUX_BASE.mio_pad_attr[mio as usize].get(),
+            Self::Dio(dio) => PINMUX_BASE.dio_pad_attr[dio as usize].get(),
+        })
+    }
+
+    // Modify value of pad attribute using common MIO/DIO register layout
+    fn modify_pad_attr(self, flags: FieldValue<u32, PAD_ATTR::Register>) {
+        let mut attr = self.pad_attr();
+        attr.modify(flags);
+        match self {
+            Self::Mio(mio) => &PINMUX_BASE.mio_pad_attr[mio as usize].set(attr.get()),
+            Self::Dio(dio) => &PINMUX_BASE.dio_pad_attr[dio as usize].set(attr.get()),
+        };
+    }
+
+    pub fn set_floating_state(&self, mode: gpio::FloatingState) {
+        self.modify_pad_attr(match mode {
+            gpio::FloatingState::PullUp => PAD_ATTR::PULL_EN::SET + PAD_ATTR::PULL::UP,
+            gpio::FloatingState::PullDown => PAD_ATTR::PULL_EN::SET + PAD_ATTR::PULL::DOWN,
+            gpio::FloatingState::PullNone => PAD_ATTR::PULL_EN::CLEAR + PAD_ATTR::PULL::CLEAR,
+        });
+    }
+
+    pub fn set_output_open_drain(self) {
+        self.modify_pad_attr(PAD_ATTR::OPEN_DRAIN_EN::SET);
+    }
+
+    pub fn set_output_push_pull(&self) {
+        self.modify_pad_attr(PAD_ATTR::OPEN_DRAIN_EN::CLEAR);
+    }
+
+    pub fn set_invert_sense(&self, invert: bool) {
+        if invert {
+            self.modify_pad_attr(PAD_ATTR::INVERT::SET)
+        } else {
+            self.modify_pad_attr(PAD_ATTR::INVERT::CLEAR)
+        }
+    }
+
+    pub fn floating_state(&self) -> gpio::FloatingState {
+        let pad_attr: PadAttribute = self.pad_attr();
+        if pad_attr.matches_all(PAD_ATTR::PULL::UP + PAD_ATTR::PULL_EN::SET) {
+            gpio::FloatingState::PullUp
+        } else if pad_attr.matches_all(PAD_ATTR::PULL::DOWN + PAD_ATTR::PULL_EN::SET) {
+            gpio::FloatingState::PullDown
+        } else {
+            gpio::FloatingState::PullNone
+        }
+    }
+
+    /// Prohibits any further changes to input/output/open-drain or pullup configuration.
+    pub fn lock_pad_attributes(&self) {
+        match *self {
+            Self::Mio(mio) => PINMUX_BASE.mio_pad_attr_regwen[(mio as u32) as usize]
+                .write(MIO_PAD_ATTR_REGWEN::EN_0::CLEAR),
+            Self::Dio(dio) => PINMUX_BASE.dio_pad_attr_regwen[(dio as u32) as usize]
+                .write(DIO_PAD_ATTR_REGWEN::EN_0::CLEAR),
+        };
+    }
+}
```

Comment:
The methods inside `impl Pad` use a mix of `&self` and `self`. I think either is fine, but please make it consistent.

Reply:
Replaced to `&self`